File: micronaut-mcp-server-java-sdk/src/test/java/io/micronaut/mcp/server/stateless/sync/completions/resources/MyResourceCompletions.java
package io.micronaut.mcp.server.stateless.sync.completions.resources;

import io.micronaut.context.annotation.Requires;
import io.micronaut.mcp.annotations.ResourceCompletion;
import jakarta.inject.Singleton;

import java.util.List;

@Requires(property = "spec.name", value = "StatelessSyncResourceCompletionsTest")
//tag::clazz[]
@Singleton
class MyResourceCompletions {
    @ResourceCompletion(uri =  "file:///home/user/documents/{fileName}")
    List<String> resourcesCompletions(String fileName) {
        return List.of(
                "report.pdf",
                "data.csv",
                "notes.txt"
            ).stream()
            .filter(name -> name.startsWith(fileName))
            .toList();
    }
}
//end::clazz[]

File: test-suite-docs-kotlin/src/test/kotlin/io/micronaut/mcp/docs/completions/resources/MyResourceCompletions.kt
package io.micronaut.mcp.docs.completions.resources

//tag::imports[]
import io.micronaut.context.annotation.Requires
import io.micronaut.mcp.annotations.ResourceCompletion
import jakarta.inject.Singleton

//end::imports[]
@Requires(property = "spec.name", value = "MyResourceCompletionsTest")
//tag::clazz[]
@Singleton
class MyResourceCompletions {
    @ResourceCompletion(uri = "file:///home/user/documents/{fileName}")
    fun resourcesCompletions(fileName: String): List<String> {
        return listOf(
                "report.pdf",
                "data.csv",
                "notes.txt"
            )
            .filter { name -> name.startsWith(fileName) }
    }
}
//end::clazz[]
